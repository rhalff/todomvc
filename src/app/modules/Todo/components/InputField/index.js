// @flow
import React, { Component } from 'react'
import styles from './scss/InputField.scss'

type Props = {
  onSubmit: Function,
  placeholder: string
}

type State = {
  value: string
}

class InputField extends Component<Props, State> {
  inputElement: ?HTMLInputElement

  state = {
    value: ''
  }

  handleSubmit = (event: SyntheticEvent<>) => {
    event.preventDefault()

    const value = this.state.value.trim()

    if (value) {
      this.props.onSubmit(value)
      this.setState({ value: '' })
    }
  }

  handleChange = () => {
    if (this.inputElement) {
      this.setState({ value: this.inputElement.value })
    }
  }

  setInputRef = (ref: ?HTMLInputElement) => {
    this.inputElement = ref
  }

  render() {
    const { placeholder } = this.props

    return (
      <form onSubmit={this.handleSubmit}>
        <input
          className={styles.edit}
          onChange={this.handleChange}
          placeholder={placeholder}
          ref={this.setInputRef}
          type="text"
          value={this.state.value}
        />
      </form>
    )
  }
}

export default InputField
