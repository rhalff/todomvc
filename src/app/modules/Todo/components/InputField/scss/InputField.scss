.inputField {
  padding: 16px 16px 16px 16px;
  border: none;
  background: transparent rgba(0, 0, 0, 0.003);
  box-shadow: inset 0 -2px 1px transparent rgba(0, 0, 0, 0.03);
}

.edit {
  composes: inputField;
  position: relative;
  margin: 0;
  width: 100%;
  font-size: 24px;
  font-family: inherit;
  font-weight: inherit;
  line-height: 1.4em;
  outline: none;
  color: inherit;
  box-sizing: border-box;
  font-smoothing: antialiased;
}
