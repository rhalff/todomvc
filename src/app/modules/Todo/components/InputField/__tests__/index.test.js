import React from 'react'
import InputField from '../'
import renderer from 'react-test-renderer'

test('Can show input field', () => {
  const component = renderer.create(
    <InputField placeholder="What needs to be done?" onSubmit={() => {}} />
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Empty input reverts to original value', () => {})
