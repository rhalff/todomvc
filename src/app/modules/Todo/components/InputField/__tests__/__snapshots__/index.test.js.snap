// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can show input field 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="edit"
    onChange={[Function]}
    placeholder="What needs to be done?"
    type="text"
    value=""
  />
</form>
`;
